DatadogInternal/Sources/NetworkInstrumentation/URLSession/DatadogURLSessionDelegate.swift

```diff
@@ -119,6 +114,37 @@ open class DatadogURLSessionDelegate: NSObject, URLSessionDataDelegate {
         // NOTE: This delegate method is only called for `URLSessionTasks` created without the completion handler.
         interceptor?.task(task, didCompleteWithError: error)
     }
+
+    private func swizzle(firstPartyHosts: FirstPartyHosts) throws {
+        try swizzler.swizzle(
+            interceptResume: { [weak self] task in
+                guard
+                    let interceptor = self?.interceptor,
+                    let provider = task.dd.delegate as? __URLSessionDelegateProviding,
+                    provider.ddURLSessionDelegate == self // intercept task with self as delegate
```

Comment:
question/ Shouldn't we use `===` here?

Reply:
Oh good catch! the `==` operator is delegating to the `NSObjectProtocol.isEqual(_:)`, so it will be the same but the `===` should be explicit here!